src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -20,11 +20,12 @@ public abstract partial class Array : ICloneable, IList, IStructuralComparable,
     {
         // We impose limits on maximum array length in each dimension to allow efficient
```

Comment:
>  We impose limits on maximum array length in each dimension to allow efficient implementation of advanced range check elimination in future.

@dotnet/jit-contrib Do you believe that artificially limiting array length to `0X7FEFFFFF` (instead of `0x7FFFFFC7` that is the natural GC limit) will be ever useful to enable these advanced range check eliminations?

We had a lot of discussion about this years ago, but none of it materialized. Having the same limit for all types of arrays would make things simpler.

Reply:
Ping on [this](https://github.com/dotnet/runtime/pull/43301#discussion_r506947747), @dotnet/jit-contrib.

> Do you believe that artificially limiting array length to 0X7FEFFFFF (instead of 0x7FFFFFC7 that is the natural GC limit) will be ever useful to enable these advanced range check eliminations?

Reply:
@dotnet/jit-contrib, can someone comment on this so we can move this PR ahead?  Thanks.

Reply:
I do not believe that the current JIT has any dependency upon these upper limits for array bounds.
It may have been done this way to support a possible induction variable optimization in the obsoleted JIT64
I scan the JIT codebases for the constant 100000 and didn't see any references to it.